server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -385,6 +388,23 @@ public PipelinedRequest resolvePipeline(SearchRequest searchRequest) throws Exce
         return new PipelinedRequest(pipeline, transformedRequest);
     }
 
+    public <Result extends SearchPhaseResult> SearchPhaseResults<Result> transformSearchPhase(
+        final SearchPhaseResults<Result> searchPhaseResult,
+        final SearchPhaseContext searchPhaseContext,
+        final String currentPhase,
+        final String nextPhase
+    ) {
+        final String pipelineId = searchPhaseContext.getRequest().pipeline();
+        if (pipelineId != null) {
+            PipelineHolder pipeline = pipelines.get(pipelineId);
+            if (pipeline == null) {
+                throw new IllegalArgumentException("Pipeline " + pipelineId + " is not defined");
+            }
+            return pipeline.pipeline.runSearchPhaseTransformer(searchPhaseResult, searchPhaseContext, currentPhase, nextPhase);
+        }
+        return searchPhaseResult;
```

Comment:
This won't work with a pipeline defined in the search request, right?

Reply:
Yes, but the time when I raised the PR, I don't think so we had that code ready. If the code is not merged I can go ahead and update this PR. @msfroh

Reply:
That code was merged last week and I just made things worse by adding https://github.com/opensearch-project/OpenSearch/pull/7597.